### ISSUE TITLE
FR: parameter similar to --runtime, but it skips 

### ISSUE THREAD

**Issue by sopsmattw** [hashcat/hashcat#2789]
Request to add a parameter similar to --runtime. maybe call it --skiptime.

The skiptime parameter takes a value in seconds for determining if the session should be skipped. This varies from runtime, which runs all sessions up to the --runtime value. This would load Time.Estimated for a session and if it's more than $skiptime, abandon the session and move on.

**Comment by Chick3nman** [comment]
Should be an easy addition, I'll see if i can knock this out this weekend. As for the flag, how about `--runtime-bypass` or `--runtime-skip`? Slightly easier to understand what it does, given what `--runtime` already does.

**Comment by sopsmattw** [comment]
I'm good with anything you decide.. You know the current thought process for parameters better than I do.

**Comment by jsteube** [comment]
I am a bit confused how @sopsmattw is using the word 'session' here. In hashcat terminology, the session is a single fixed name that does not change no matter which position inside the attack queue we are in. If it's not specified by the user, it is typically just 'hashcat'. Or is it that we are actually talking about sessions and I just don't understand it?

**Comment by Chick3nman** [comment]
It's not talking about sessions. We spoke out of band about this issue and I asked him to make an issue here so i don't forget to do it. Feel free to disregard until i PR.

**Comment by sopsmattw** [comment]
The confusion is likely because I'm wielding terms completely in the wrong way. :) The idea is to mimic the functionality of --runtime but instead running until that timer triggers, skip the current iteration if the Time.Estimated exceeds the passed parameter value.  i.e. if 9 chr brute force is going to take longer than x seconds, skip 9chr altogether.

**Comment by jsteube** [comment]
Yeah I got that, it's a nice idea. 

@Chick3nman I think the easiest way would be to call the function which is called in case the user asks for bypass from command prompt or mimic it. We need to stick close to this because there are some details in regards to return codes on the shell which H2P checks, for instance. Also the ETA shouldn't be checked before the first checkpoint was reached. Probably the best way here it to run an additional new thread the same way how runtime does it.

**Comment by Chick3nman** [comment]
Yeah, the plan was to wait for a timer or checkpoint to allow speeds to stabilize, check the estimated time, and then issue bypass if the time is too long. Should be very easy to do, i just haven't had a chance to do it since friday.
